mathewDavidov/assignments/slidingwindow/lc209/lc209.java

```diff
@@ -0,0 +1,52 @@
+/*
+Given an array of positive integers nums and a positive integer target, 
+return the minimal length of a contiguous subarray [numsl, numsl+1, ..., numsr-1, numsr] 
+of which the sum is greater than or equal to target. If there is no such subarray, return 0 instead.
+
+Input: target = 7, nums = [2,3,1,2,4,3]
+Output: 2
+Explanation: The subarray [4,3] has the minimal length under the problem constraint.
+
+edge:
+arr.length > 0 || null
+target > 0
+
+currentSum, windowStart, minLength
+for 0 - end
+    currentSum += arr[windowEnd]
+
+    while currentSum >= target
+        minLength = min(minLength, windowEnd-windowStart+1)
+        currentSum -= arr[windowStart]
+        windowStart++;
+*/
+
+import java.util.*;
+
+public class lc209 {
+    public static int findMinSubArray(int target, int[] arr) {
+        if (arr == null || arr.length == 0) {
+            return 0;
+        }
+
+        int minLength = arr.length, currentSum = 0, windowStart = 0;
+
+        for (int windowEnd = 0; windowEnd < arr.length; windowEnd++) {
+            currentSum += arr[windowEnd];
+
+            while (currentSum >= target) {
+                minLength = Math.min(minLength, windowEnd - windowStart + 1);
+                currentSum -= arr[windowStart];
+                windowStart++;
+            }
+        }
+
+        return minLength;
```

Comment:
In this solution, if you don't find the target sum in the array, it wouldn't return 0, it would return the length of the array. You need to check whether the minLength is less than the array's length first and if not, return 0 because it was never changed and the sum was never found. You also probably need to initialize minLength as the array's length +1 because what if the shortest subArray was actually the entire array?